services/ui-src/src/views/Profile/Profile.tsx

```diff
@@ -1,8 +1,82 @@
 // components
-import { Flex, Text } from "@chakra-ui/react";
+import { Button, Flex, Heading, Text } from "@chakra-ui/react";
+import { RouterLink } from "../../components/index";
+//utils
+import { useUser } from "utils/auth";
 
-export const Profile = () => (
-  <Flex h="100%" justifyContent="center" py="12">
-    <Text data-testid="profile">This is the profile. More to come later.</Text>
-  </Flex>
-);
+const userDetails = () => {
+  const userInfo = useUser();
+  const { email, given_name, family_name } = userInfo.user.attributes;
+  const { userRole } = userInfo;
+  const state = userInfo.user.attributes?.["custom:cms_state"] || "";
+  return { email, given_name, family_name, userRole, state };
+};
```

Comment:
Could just move this inside the component itself and not have a separate function. Could go either way though. 🤷

Reply:
Personally I like separating it so that rendering the component is somewhat isolated and the way we fetch info can change. Maybe this is too granular.